Redesign subtopic viewer

shavavo opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->
This PR is based off of PR #9362 and should be merged only after PR #9362 is merged. Please review commits starting from commit 551450a

1. This PR does the following: Redesigns subtopic viewer as [here](https://xd.adobe.com/view/83e97e1d-7820-4387-72b2-8860a2ee8f6d-f5d8/screen/582b3bcd-ac57-4442-ad92-7fe0c4f7f0c4/Revision-Player-Naming-numbers/)

## Screenshots
1. Web
![Screen Shot 2020-06-02 at 1 32 12 PM](https://user-images.githubusercontent.com/33111844/83550900-8e438380-a4d5-11ea-885c-0d3d2c32cb78.png)



2. Mobile

![Screen Shot 2020-06-02 at 1 32 27 PM](https://user-images.githubusercontent.com/33111844/83550912-93083780-a4d5-11ea-81a4-d5d3cab943bd.png)





## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

aks681 committed:
Added topic info tab and redesigned tabs

aks681 committed:
Done with info tab

aks681 committed:
started with stories tab

aks681 committed:
revamped story tab

aks681 committed:
disable flag

aks681 committed:
made review changes

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into topic-viewer

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into topic-viewer

aks681 committed:
mobile topic info tab done

aks681 committed:
story tab mobile done

aks681 committed:
Done with practice sessions tab

aks681 committed:
Finsihed subtopics tab and edits to story tab

Commit:
Merge branch 'topic-viewer' of https://github.com/aks681/oppia into aks681-topic-viewer

Commit:
redesign subtopic viewer

Commit:
hooked up new subtopic viewer

Commit:
Added translation value

Commit:
lint fixes

Commit:
more lint errors

Commit:
fixed python linting

Commit:
forgot comma

Commit:
fixed test

Commit:
tests for ReadOnlySubtopicPageObjectFactory.ts

Commit:
frontend checks

Commit:
frontend checks

Commit:
Merge remote-tracking branch 'upstream/develop' into redesign-subtopic-viewer-off-aks681-topic-viewer

shavavo commented:
@kevintab95  @mschanteltc  PTAL

seanlip commented on assets/i18n/en.json:
```diff
@@ -526,6 +526,7 @@
     "I18N_START_HERE": "Click here to start!",
     "I18N_STORY_VIEWER_COMPLETED_CHAPTER": "<[title]> - Completed!",
     "I18N_SUBSCRIBE_BUTTON_TEXT": "Subscribe",
+    "I18N_SUBTOPIC_VIEWER_NEXT_SKILL": "Next Skill:",
```
Probably don't include the comma here.

shavavo replied:
Done.

seanlip commented on core/controllers/subtopic_viewer.py:
```diff
@@ -59,17 +59,22 @@ def get(self, topic_name, subtopic_id):
             if subtopic.id == subtopic_id:
                 subtopic_title = subtopic.title
+                if index != len(topic.subtopics) - 1:
+                    next_subtopic = topic.subtopics[index + 1].to_dict()
```
Naming is wrong -- this suggests a domain object. Probably you'll want it to be next_subtopic_dict.

Do you need to pass the whole subtopic, though? If possible, it would be better to pass just the minimal subset of data required.

shavavo replied:
I need to construct a subtopic object with the dictionary, so would probably need all of it

shavavo replied:
Done.

seanlip commented on core/templates/domain/subtopic_viewer/ReadOnlySubtopicPageObjectFactory.ts:
```diff
@@ -25,19 +25,36 @@ import {
+      parentTopicId: string,
+      subtopicTitle: string,
+      pageContents: SubtopicPageContents,
+      nextSubtopic: Subtopic| null
```
For consistency, add space before | too.

shavavo replied:
Done.

seanlip commented on core/templates/pages/subtopic-viewer-page/subtopic-viewer-page.controller.ts:
```diff
@@ -66,6 +68,15 @@ angular.module('oppia').directive('subtopicViewerPage', [
                 PageTitleService.setPageTitle(ctrl.subtopicTitle + ' - Oppia');
+
+                let nextSubtopic =
+                  subtopicDataObject.getNextSubtopic();
```
Use parens to clearly indicate that the next line is a continuation: 

```
    let nextSubtopic = (
      subtopicDataObject.getNextSubtopic());
```

shavavo replied:
Done.

seanlip commented on scripts/check_frontend_coverage.py:
```diff
@@ -459,6 +459,7 @@
     'topic-editor-state.service.ts',
     'topic-editor-stories-list.directive.ts',
     'topic-editor-tab.directive.ts',
+    'topic-info-tab.directive.ts',
```
You should not be adding any files to this list. See the note at the top of the list. /cc @marianazangrossi

shavavo replied:
My bad on that, I was not passing checks due to I believe akshay's changes. Is there a way to push with skipping coverage tests?

seanlip replied:
No, we strongly discourage pushing in that scenario. Talk to me if you need to do that.

shavavo replied:
Reverted

shavavo replied:
Done.

seanlip reviewed (changes requested):
Thanks @shavavo, took a pass!

seanlip commented:
Adding @mschanteltc for UI review.

Also, @shavavo -- the image at the bottom right in the screenshot seems to be missing, is that expected?

mschanteltc commented:
@shavavo 

This looks great so far. Here are my suggestions/questions:

- For the green color used in the header and Subtopic name, use #00645C
- What is the width of the contents in the white and blue container? And do they match the padding and font size of that of the [mocks](https://xd.adobe.com/view/83e97e1d-7820-4387-72b2-8860a2ee8f6d-f5d8/screen/652f5ede-7fcb-4273-851f-dda6f5e38e27/Revision-Player-Sorting-Numbers)?  I try to aim for 12~15 words a line but something seems different here.
- The curved corners should have a radius of 4 px. 
- "Next Skill" should use Roboto font
- The Revision card names should use medium weight Roboto font
- The orange color used in the Revision card is #BE563C

aks681 committed:
Addressed sean's comments

aks681 committed:
addressed UI comments and other changes

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into topic-viewer

aks681 committed:
minor change

Commit:
cleaned up

Commit:
formatting

shavavo commented:
@mschanteltc Is the color of the header bar a global change we want to make? Old value: #009688, new value: #00645C

Also, I will forward the revision card changes to akshay
Edit: nvm, I see that it's already there on his PR. Once it's updated there, it should update here next time I push

shavavo commented:
@seanlip I just didn't add a svg, but it is usually required

Commit:
closer to mockups

shavavo commented:
@mschanteltc updated screenshots PTAL

mschanteltc commented:
> @mschanteltc updated screenshots PTAL

Thanks for the update! And yes, the color change should be global. Here are my comments about the mocks:

- The white container outside of "Rounding numbers" should have a faint drop shadow. The one I used in my mocks is centered with a radius of 4 px. It is #000000 at 10% opacity. Although the drop shadow does not have to match what I described, it should be identical to the drop shadows used in the white containers of all the other pages (Info Tab, Lessons Tab, Chapter List, etc.)
- The Next Skill card should be right aligned to the blue container with the skill content. The "Next Skill" heading is still left aligned, which the mock shows, but the grouping needs correct alignment.

Question: Is someone else developing the Skill Card? It's being used in Topic Page's Revision Tab and I'm not sure who I should be giving feedback to without sounding repetitive.

seanlip reviewed (approved):
Thanks @shavavo! No further comments from my end.

seanlip commented:
Your frontend tests are failing though. (Did they not run locally prior to push on the pre-push script? If not, please consult with @Hudda to debug.)

aks681 commented on core/templates/pages/subtopic-viewer-page/subtopic-viewer-page.controller.ts:
```diff
@@ -66,6 +68,15 @@ angular.module('oppia').directive('subtopicViewerPage', [
+                let nextSubtopic = (
+                  subtopicDataObject.getNextSubtopic());
+                if (nextSubtopic) {
+                  ctrl.parentTopicId = subtopicDataObject.getParentTopicId();
```
Why is this set, only when nextSubtopic is present? Shouldn't a back button to the topic viewer exist in all cases (I don't see that in the screenshot as well)?

shavavo replied:
I need topic id to pass it down into subtopic-summary-tile, I believe

aks681 replied:
Ah, ok, yeah I just checked, it is required to display the thumbnail image. Though, my second question still stands, shouldn't there be a back button to the topic viewer as seen  in the mocks?

shavavo replied:
You are correct, I'll add that

shavavo replied:
Done.

shavavo replied:
Done.

Commit:
Merge branch 'topic-viewer' of https://github.com/aks681/oppia into redesign-subtopic-viewer-off-aks681-topic-viewer

Commit:
design

Commit:
Styling

Commit:
Added back button

Commit:
trailing space

shavavo commented:
@mschanteltc PTAL, updated screenshots. I added a back button as well, but couldn't get it to be the same padding as the mockup without making big changes to how navigation bar works globally- Do you think that's okay or is it a little cramped?

Commit:
formatting

Commit:
styling

shavavo commented:
@aks681 PTAL. @marianazangrossi I created a new file in this commit (subtopic-viewer-navbar-pre-logo-action.directive.ts), could I add it to NOT_FULLY_COVERED_FILENAMES?

seanlip commented:
> @aks681 PTAL. @marianazangrossi I created a new file in this commit (subtopic-viewer-navbar-pre-logo-action.directive.ts), could I add it to NOT_FULLY_COVERED_FILENAMES?

In general, no, since the aim of that list is to ensure that newly-added files do get tested. (Though @marianazangrossi makes the final call here.)

seanlip commented:
@shavavo could you please update the assignees list accordingly whenever you want someone else to take a look at a PR?

seanlip commented:
(It might be worth adding codeowners too, if you think it's ready for them.)

kevintab95 commented:
Please note that frontend tests are failing due to coverage issues:
```
------------------------------------
Frontend Coverage Checks Not Passed.
------------------------------------
subtopic-viewer-navbar-pre-logo-action.directive.ts seems to be not completely tested. Make sure it's fully covered.
topic-info-tab.directive.ts seems to be not completely tested. Make sure it's fully covered.
```

kevintab95 commented on core/templates/pages/subtopic-viewer-page/navbar-breadcrumb/subtopic-viewer-navbar-breadcrumb.directive.html:
```diff
@@ -1,10 +1,7 @@
-  <span>
-    <[subtopicTitle]>
-  </span>
+<div ng-if="subtopicTitle" class="nav navbar-nav oppia-navbar-breadcrumb ng-cloak" style="padding-top: 0; align-items: center; height: 56px;">
```
Use classes here and below. Avoid inline styling.

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/subtopic-viewer-page/navbar-pre-logo-action/subtopic-viewer-navbar-pre-logo-action.directive.html:
```diff
@@ -0,0 +1,5 @@
+  <span class="material-icons oppia-navbar-back-button">
+    arrow_back
+  </span>
+</a>
```
Add a new line in line 6.

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/subtopic-viewer-page/navbar-pre-logo-action/subtopic-viewer-navbar-pre-logo-action.directive.ts:
```diff
@@ -0,0 +1,48 @@
+angular.module('oppia').directive('subtopicViewerNavbarPreLogoAction', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
```
Avoid creating new directives. We're moving directives to components. PTAL at #9317 for reference.

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/subtopic-viewer-page/subtopic-viewer-page.directive.html:
```diff
@@ -1,42 +1,118 @@
+    margin-left: 50px;
+    margin-right: 50px;
+    margin-top: 4vh;
+    padding-bottom: 40px;
+    padding-left: 50px;
+    padding-right: 50px;
+    padding-top: 40px;
```
```suggestion
    padding: 40px 50px;
```

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/subtopic-viewer-page/subtopic-viewer-page.directive.html:
```diff
@@ -1,42 +1,118 @@
+    }
+
+    .oppia-page-contents-viewer {
+      padding-bottom: 30px;
+      padding-left: 20px;
+      padding-right: 20px;
+      padding-top: 30px;
```
```suggestion
      padding: 30px 20px;
```

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/subtopic-viewer-page/subtopic-viewer-page.directive.html:
```diff
@@ -1,42 +1,118 @@
-      </angular-html-bind>
+
+  <div ng-if="$ctrl.nextSubtopicSummaryIsShown" class="oppia-subtopic-viewer-next-subtopic-summary">
+    <div style="margin-right: 50px; margin-top: 80px;">
```
Avoid inline styling here.

shavavo replied:
Done.

kevintab95 commented on core/templates/pages/topic-viewer-page/info-tab/topic-info-tab.directive.html:
```diff
@@ -0,0 +1,140 @@
+    <div class="info-tab-content">
+      <div class="info-tab-left-container" ng-if="!$ctrl.isMobile">
+        <md-card class="info-tab-count">
+          <div style="padding-top: 12%;">
```
Avoid inline styling here and elsewhere in this file.

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/info-tab/topic-info-tab.directive.ts:
```diff
@@ -0,0 +1,50 @@
+angular.module('oppia').directive('topicInfoTab', ['UrlInterpolationService',
+  function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
```
Avoid creating new directives. Use components instead. See #9317 for reference.

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/practice-tab/practice-tab.directive.html:
```diff
@@ -1,31 +1,68 @@
+       but since those CSS classes are evaluated after the classes in this directive,
+       without the !important tag, any styles set here will be ignored for common classes.
+    */
+    width: 37.5vw !important;
```
Avoid using `!important` (see [guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#css)). Use the concept of CSS specificity to style the correct element. You can create new classes if required.

kevintab95 replied:
Ditto below.

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/practice-tab/practice-tab.directive.html:
```diff
@@ -1,31 +1,68 @@
-    width: 100%;
-    max-width: 800px;
+    border-radius: 0.4em;
+    font-family: "Roboto", Arial, sans-serif;
```
Remove quotes here if it is not required.

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/practice-tab/practice-tab.directive.html:
```diff
@@ -1,31 +1,68 @@
-    height: 100px;
-    text-align: center;
+  practice-tab .practice-content {
+    font-family: "Roboto", Arial, sans-serif;
```
Ditto: quotes.

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/topic-viewer-page.directive.html:
```diff
@@ -1,54 +1,97 @@
+
   .oppia-topic-viewer-tabs-container {
+    background-color: #00645C;
+    font-family: 'Capriola', 'Roboto', Arial, sans-serif;
```
Ditto: quotes

shavavo replied:
@aks681

kevintab95 commented on core/templates/pages/topic-viewer-page/topic-viewer-page.directive.html:
```diff
@@ -61,23 +104,32 @@
+
+  .topic-viewer-page-content .tab-title {
+    color: #01645c;
+    font-family: 'Capriola', 'Roboto', Arial, sans-serif;
```
ditto: quotes.

shavavo replied:
@aks681

kevintab95 reviewed (changes requested):
Thanks @shavavo! I took a pass and left some comments. PTAL!

vojtechjelinek commented on core/templates/base-components/base-content.directive.ts:
```diff
@@ -29,9 +29,12 @@ angular.module('oppia').directive('baseContent', [
       restrict: 'E',
       scope: {},
-      bindToController: {},
+      bindToController: {
+        backButtonShown: '<'
+      },
       transclude: {
         breadcrumb: '?navbarBreadcrumb',
+        preLogoAction: '?navbarPreLogoAction',
```
Why do you need to modify the base components? Can this be done is some other file, I assume this is only used in the new subtopic viewer and not in the other places.

shavavo replied:
This is for adding a back button for the nav header. I thought it could be utilized on other pages in the future.

vojtechjelinek replied:
Okay, makes sense.

vojtechjelinek reviewed (changes requested):
Thanks! One comment.

Commit:
moved subtopic-viewer-navbar-pre-logo-action to component, added tests

shavavo commented:
@kevintab95 some of your comments were directed at changes on PR #9362, I've tagged akshay in the ones that apply. PTAL @kevintab95, @vojtechjelinek

mschanteltc commented:
> @mschanteltc PTAL, updated screenshots. I added a back button as well, but couldn't get it to be the same padding as the mockup without making big changes to how navigation bar works globally- Do you think that's okay or is it a little cramped?

Thanks @shavavo! I do think that the back button is cramped and can use some more space on its right side. @seanlip Correct me if I'm mistaken, but I thought that globally, all navbars would either have a hamburger menu or a back button to the left of the logo?

In regards to the skill card, is this being created here or by someone else? I have several comments to make about how it is formatted but I know that there are other instances of this card on other pages so I'm not sure who I should provide feedback to.

vojtechjelinek reviewed (approved):
LGTM!

seanlip commented:
@mschanteltc Currently all navbars only have a hamburger menu in mobile, and no menu or back button in desktop. We typically never use the Back button -- instead, we use breadcrumbs (e.g. Fractions > Adding Fractions).

I defer to you on the design questions but the above is my understanding of the current state of things. (Does it address your question?)

seanlip commented:
(Also @mschanteltc, please add me to the Assignees list on the right of the page if you'd like me to take the next action on the thread, and remove yourself from it if the ball isn't in your court. Please follow this principle for all github threads. Thanks!)

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into topic-viewer

aks681 committed:
addressed comments

aks681 committed:
added directive test

aks681 committed:
fixed typescript error

shavavo commented:
@mschanteltc You can direct feedback for the skill summary tiles to this PR: #9362

aks681 committed:
fixed backend errors

Commit:
fix for top nav bar; better spacing on back button

Commit:
updated test

shavavo commented:
@mschanteltc I've updated the nav bar spacing - PTAL. Here's the screenshot
![Screen Shot 2020-06-04 at 12 25 15 PM](https://user-images.githubusercontent.com/33111844/83785012-abf02480-a65e-11ea-848f-f07a64f5b6fa.png)

Commit:
Merge branch 'topic-viewer' of https://github.com/aks681/oppia into redesign-subtopic-viewer-off-aks681-topic-viewer

kevintab95 commented on core/templates/pages/topic-viewer-page/info-tab/topic-info-tab.directive.ts:
```diff
@@ -38,11 +48,15 @@ angular.module('oppia').directive('topicInfoTab', ['UrlInterpolationService',
-              ctrl.isMobile = true;
+              return true;
             }
+            return false;
```
Can you change this to:

`return WindowDimensionsService.getWidth() <= 1024;`

shavavo replied:
@aks681

shavavo replied:
This change should go in #9362 @aks681

kevintab95 commented on core/templates/pages/subtopic-viewer-page/navbar-breadcrumb/subtopic-viewer-navbar-breadcrumb.directive.html:
```diff
@@ -1,7 +1,27 @@
+    display: flex;
+    flex-direction: column;
+  }
+</style>
```
Add a new line after this.

shavavo replied:
Done.

kevintab95 reviewed (changes requested):
Thanks @shavavo! This looks great. I have just a few nits. PTAL!

Commit:
styling

shavavo commented:
@kevintab95 I've forward your first comment to #9362, and fixed the 2nd comment. PTAL

kevintab95 reviewed (approved):
LGTM, thanks @shavavo!

mschanteltc commented:
> @mschanteltc Currently all navbars only have a hamburger menu in mobile, and no menu or back button in desktop. We typically never use the Back button -- instead, we use breadcrumbs (e.g. Fractions > Adding Fractions).
> 
> I defer to you on the design questions but the above is my understanding of the current state of things. (Does it address your question?)

Yes, although I would like to know which pages would not have any button? 
Also thank you for reminding me of that principle to follow! I'll stay assigned to this PR until all the design parts are complete.

seanlip commented:
> Yes, although I would like to know which pages would not have any button?

I'm not sure what you mean? Currently, no pages have Back buttons. Introducing a Back button is a new design pattern, and if we're going to do that, I think it's up to you which pages should have them or not.

nishantwrp reviewed (approved):
lgtm as a codeowner!

shavavo commented:
@ankita240796 @DubeySandeep @marianazangrossi PTAL

ankita240796 reviewed (approved):
LGTM from codeowner's perspective, Thanks @shavavo!

Commit:
qqq.json

mschanteltc commented:
> > Yes, although I would like to know which pages would not have any button?
> 
> I'm not sure what you mean? Currently, no pages have Back buttons. Introducing a Back button is a new design pattern, and if we're going to do that, I think it's up to you which pages should have them or not.

What I mean to say is if there are pages that have no navigation button on the left of the header, the new design would require all headers to either have a hamburger menu or a back button.

seanlip commented:
OK! I'm fine with that, I think which pages it applies to is your call.

DubeySandeep commented on main.py:
```diff
@@ -263,10 +263,12 @@ def ui_access_wrapper(self, *args, **kwargs):
     get_redirect_route(
-        r'%s/<topic_name>' % feconf.PRACTICE_SESSION_URL_PREFIX,
+        r'%s/<topic_name>' %
+        feconf.PRACTICE_SESSION_URL_PREFIX,
```
Any specific reason for this change?

shavavo replied:
Reverted - Done

DubeySandeep commented on main.py:
```diff
@@ -263,10 +263,12 @@ def ui_access_wrapper(self, *args, **kwargs):
     get_redirect_route(
-        r'%s/<topic_name>' % feconf.PRACTICE_SESSION_DATA_URL_PREFIX,
+        r'%s/<topic_name>' %
+        feconf.PRACTICE_SESSION_DATA_URL_PREFIX,
```
Ditto

shavavo replied:
Reverted - Done

DubeySandeep reviewed (changes requested):
LGTM, one minor comment PTAL!

jameesjohn reviewed (approved):
LGTM for codeowner files

marianadasilvadev reviewed (approved):
LGTM as codeowner perspective

Commit:
MC

Commit:
Styling

Commit:
styling

Commit:
object factory upgraded services

shavavo commented:
@DubeySandeep PTAL and merge for release

vojtechjelinek commented:
@shavavo The coverage is not not 100%, please fix that.

bansalnitish reviewed (approved):
LGTM for code owner's file (UpgradedServices.ts)!

nithusha21 reviewed (approved):
LGTM as a codeowner

aks681 reviewed (commented):
Lgtm, thanks! Though one comment, in the backup server, it was seen that the spacing wasn't right between paragraphs in the subtopic viewer, can you just verify, and if not, fix the spacing issue in the subtopic viewer?
![Screenshot from 2020-06-08 02-36-11](https://user-images.githubusercontent.com/22347970/84018337-8edc8e00-a99d-11ea-82d8-31d676abdd46.png)

Commit:
increased subtopic viewer test coverage

shavavo commented:
@aks681 It looks like that screenshot was taken on the old version of subtopic viewer - though little should have changed in regards to the content layout. I tried reproducing the issue with the redesign and was unsuccessful:

![Screen Shot 2020-06-08 at 11 34 47 AM](https://user-images.githubusercontent.com/33111844/84050804-3943b980-a97c-11ea-930d-097dcfe3b859.png)

aks681 reviewed (approved):
Lgtm!

DubeySandeep reviewed (approved):
LGTM codeowner files

Commit:
Merge branch 'develop' into redesign-subtopic-viewer-off-aks681-topic-viewer

kevintab95 commented:
There was a merge conflict after #9392 got merged. I have resolved the conflict because this PR is a bit time-sensitive and needs to be in the current release. Also, since the last commit was a minor change, we can merge this PR after some of the basic tests pass. Thanks!

mschanteltc commented:
> @mschanteltc I've updated the nav bar spacing - PTAL. Here's the screenshot
> ![Screen Shot 2020-06-04 at 12 25 15 PM](https://user-images.githubusercontent.com/33111844/83785012-abf02480-a65e-11ea-848f-f07a64f5b6fa.png)

Yes this looks good, thank you!
